packages/frontpage/app/api/receive_hook/route.ts

```diff
@@ -31,113 +21,42 @@ export async function POST(request: Request) {
   }
 
   const { ops, repo, seq } = commit.data;
+  const row = await db
+    .select()
+    .from(schema.ConsumedOffset)
+    .where(eq(schema.ConsumedOffset.offset, seq))
+    .limit(1);
+
+  const operationConsumed = Boolean(row[0]);
+  if (operationConsumed) {
+    console.log("Already consumed sequence:", seq);
+    return new Response("OK");
+  }
+
   const service = await getPdsUrl(repo);
   if (!service) {
     throw new Error("No AtprotoPersonalDataServer service found");
   }
-
   const promises = ops.map(async (op) => {
```

Comment:
Note for later: this shouldn't be a Promise.all. Can lead to race conditions if a commit has multiple ops in. We can fix in another PR.